ex198a is

  f =>
    say "in f, perform exit:"
    x := u8 4
    exit x
    say "** not reachable code **"

  my_exit : Exit_Handler is
    redef exit(code u8) void =>
      io.err.println "*** user defined exit handler diverting exit $code ***"
      exit.type.abort

  exit my_exit ()->
    f
    say "** not reachable code **"

  say "exit normally"
